components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -627,7 +689,115 @@ private String replaceGroupNamesWithId(String permissionString) throws ParseExce
         } catch (IdentityProviderException e) {
             String errorMessage = "There are invalid roles in the permission string";
             log.error(errorMessage, e);
-            throw new APIManagementException(errorMessage, ExceptionCodes.UNSUPPORTED_ROLE);
+            throw new APIManagementException(errorMessage, e, ExceptionCodes.UNSUPPORTED_ROLE);
+        }
+        return updatedPermissionArray.toJSONString();
+    }
+
+    /**
+     * This method retrieves the set of overall permissions for a given api for the logged in user
+     *
+     * @param loggedInUserName - Logged in user
+     * @param api              - The API whose permissions for the logged in user is retrieved
+     * @return The overall list of permissions for the given API for the logged in user
+     */
+    private List<String> getAPIPermissionsOfLoggedInUser(String loggedInUserName, API api)
+            throws APIManagementException {
+        Set<String> permissionArrayForUser = new HashSet<>();
+        Map<String, Integer> permissionMap = api.getPermissionMap();
+        String provider = api.getProvider();
+        //TODO: Remove the check for admin after IS adds an ID to admin user
+        if (loggedInUserName.equals(provider) || permissionMap == null || permissionMap.isEmpty() || "admin"
```

Comment:
This check is made because of a restriction on the default user admin in SCIM. The hardcoded admin username can be removed after the issue reported in [1] is resolved.

[1] https://github.com/wso2/carbon-apimgt/issues/4307